finder/finder.go

```diff
@@ -1,60 +1,76 @@
-package main
+package finder
 
 import (
 	"errors"
 	"fmt"
+	"log"
 	"sync"
+
+	"github.com/status-im/go-maven-resolver/fetcher"
+	"github.com/status-im/go-maven-resolver/pom"
 )
 
 type Finder struct {
 	deps         map[string]bool /* to avoid checking the same dep */
-	mtx          sync.Mutex      /* for locking access to the deps map */
-	wg           sync.WaitGroup  /* to figure out when it's done */
-	fetchers     FetcherPool     /* pool of workers for HTTP requests */
+	fetchers     fetcher.Pool    /* pool of workers for HTTP requests */
 	ignoreScopes []string        /* list of scopes to ignore */
 	recursive    bool            /* recursive resolution control */
+	l            *log.Logger     /* for logging events */
+
+	mtx sync.Mutex     /* for locking access to the deps map */
+	wg  sync.WaitGroup /* to figure out when it's done */
 }
 
-func (f *Finder) ResolveDep(dep Dependency) (string, *Project, error) {
-	var rval *FetcherResult
+func New(deps map[string]bool, fetchers fetcher.Pool, ignoreScopes []string, recursive bool, logger *log.Logger) Finder {
+	return Finder{
+		deps:         deps,
+		fetchers:     fetchers,
+		ignoreScopes: ignoreScopes,
+		recursive:    recursive,
+		l:            logger,
+	}
+}
+
+func (f *Finder) ResolveDep(dep pom.Dependency) (string, *pom.Project, error) {
+	var rval *fetcher.Result
 	var repo string
-	result := make(chan *FetcherResult)
+	result := make(chan *fetcher.Result)
 	defer close(result)
 
 	if !dep.HasVersion() {
 		path := dep.GetMetaPath()
 		/* We use workers for HTTP request to avoid running out of sockets */
-		f.fetchers.queue <- &FetcherJob{result, path, repo}
+		f.fetchers.Queue <- &fetcher.Job{result, path, repo}
```

Comment:
This could be changed to `fetch.NewJob(result, path, repo)` so that `fetcher.Job` doesn't need to export its fields.